internal/pkg/service/objectstorage/cloud/azure/storage.go

```diff
@@ -61,120 +54,38 @@ func NewAzureStorageService(storageAccountClient *armstorage.AccountsClient, blo
 // If the storage account exists, it then checks if the BlobContainer with the specified name exists in the storage account.
 // If the BlobContainer does not exist, it returns false. Otherwise, it returns true.
 func (s AzureObjectStorageAdapter) ExistsBucket(ctx context.Context, bucket *v1alpha1.Bucket) (bool, error) {
+	storageAccountName := sanitizeStorageAccountName(bucket.Spec.Name)
+
 	// Check if storage account exists on Azure
-	existsStorageAccount, err := s.existsStorageAccount(ctx, bucket.Spec.Name)
+	existsStorageAccount, err := s.existsStorageAccount(ctx, storageAccountName)
 	if err != nil {
 		return false, err
 	}
 	// If StorageAccount does not exists that means the bucket does not exists too, so we return false
 	if !existsStorageAccount {
 		return false, nil
 	}
-	// Check BlobContainer name exists in StorageAccount
-	_, err = s.blobContainerClient.Get(
-		ctx,
-		s.cluster.GetResourceGroup(),
-		s.getStorageAccountName(bucket.Spec.Name),
-		bucket.Spec.Name,
-		nil)
-	if err != nil {
-		var respErr *azcore.ResponseError
-		if errors.As(err, &respErr) {
-			// If NOT FOUND error, that means the BlobContainer doesn't exist, so we return false
-			if respErr.StatusCode == http.StatusNotFound {
-				return false, nil
-			}
-		}
-		return false, err
-	}
-
-	return true, nil
-}
 
-// existsStorageAccount checks if a storage account exists for the given bucket name in Azure Object Storage.
-// It returns a boolean indicating whether the storage account exists or not, along with any error encountered.
-func (s AzureObjectStorageAdapter) existsStorageAccount(ctx context.Context, bucketName string) (bool, error) {
-	availability, err := s.storageAccountClient.CheckNameAvailability(
-		ctx,
-		armstorage.AccountCheckNameAvailabilityParameters{
-			Name: to.Ptr(s.getStorageAccountName(bucketName)),
-			Type: to.Ptr("Microsoft.Storage/storageAccounts"),
-		},
-		nil)
-	if err != nil {
-		return false, err
-	}
-	return !*availability.NameAvailable, nil
+	return s.existsContainer(ctx, bucket, storageAccountName)
 }
 
-// CreateBucket creates the Storage Account if it not exists AND the Storage Container
+// / CreateBucket creates the Storage Account if it not exists AND the Storage Container
 // CreateBucket creates a bucket in Azure Object Storage.
 // It checks if the storage account exists, and if not, it creates it.
 // Then, it creates a storage container within the storage account.
 // Finally, it retrieves the access key for 'key1' and creates a K8S Secret to store the storage account access key.
 // The Secret is created in the same namespace as the bucket.
 // The function returns an error if any of the operations fail.
 func (s AzureObjectStorageAdapter) CreateBucket(ctx context.Context, bucket *v1alpha1.Bucket) error {
```

Comment:
This should rather be named `UpsertBucket`

Reply:
Not really because we have no need to update the aws s3 bucket https://github.com/giantswarm/object-storage-operator/pull/261/files/6ec3ad5bb295b912ea42d146e2523ec5d4d6fd04#diff-7c520fc74cf4bc44bb13c84b9b791d343556214b519571b055104cd130e3094bR68

Reply:
I get that Azure can do update and AWS can't but this function is called from UpdateBucket and also contains functions named upsert, it would make things simpler to just call this UpsertBucket and use it instead of Create and Update

Reply:
But then we need to make sure we do not update when we're in azure and that is a major overhaul of the whole codebase

Reply:
ok fine, I though it would make things easier a not more complex.

Reply:
It might make it easier for sure but my goal was to reduce the number of changes. I can try to do it once the other pr is merged